containers/src/Data/IntMap/Internal.hs

```diff
@@ -3322,17 +3338,23 @@ fromSet f (IntSet.Tip kx bm) = buildTree f kx bm (IntSet.suffixBitMask + 1)
     -- create a Bin node, otherwise exactly one of them is nonempty
     -- and we construct the IntMap from that half.
     buildTree g !prefix !bmask bits = case bits of
-      0 -> Tip prefix (g prefix)
+      0 -> Tip prefix <$> (g prefix)
       _ -> case bits `iShiftRL` 1 of
         bits2
           | bmask .&. ((1 `shiftLL` bits2) - 1) == 0 ->
               buildTree g (prefix + bits2) (bmask `shiftRL` bits2) bits2
           | (bmask `shiftRL` bits2) .&. ((1 `shiftLL` bits2) - 1) == 0 ->
               buildTree g prefix bmask bits2
           | otherwise ->
-              Bin (Prefix (prefix .|. bits2))
-                (buildTree g prefix bmask bits2)
-                (buildTree g (prefix + bits2) (bmask `shiftRL` bits2) bits2)
+              liftA2
+                (Bin (Prefix (prefix .|. bits2)))
+                  (buildTree g prefix bmask bits2)
+                  (buildTree g (prefix + bits2) (bmask `shiftRL` bits2) bits2)
+#if __GLASGOW_HASKELL__
+{-# INLINABLE fromSetA #-}
+#else
+{-# INLINE fromSetA #-}
+#endif
```

Comment:
Why `INLINE` on non-GHC?

Reply:
I presume that there is a reason this pattern is used 45 other times for other functions; I didn't want to break anything by using INLINABLE when it isn't able to be used.

Reply:
They seem to have originated in https://github.com/haskell/containers/commit/06e91accde415ccdf1a680d9436bf2d83731183b, where INLINABLE was added for specialization on GHC. There is really no reason to do this today. Perhaps we should clean these up to avoid confusion. Anyway, it should be harmless to let it be for now.